app/src/main/java/com/nervousfish/nervousfish/activities/CreateProfileHelper.java

```diff
@@ -26,21 +26,48 @@
      * containing its logical functionality.
      *
      * @param keyGenerator A {@link IKeyGenerator} to generate keys.
-     * @param alertColor The {@link Color} to set as alert for incorrect values.
+     * @param alertColor   The {@link Color} to set as alert for incorrect values.
      */
     CreateProfileHelper(final IKeyGenerator keyGenerator, final int alertColor) {
         this.keyGenerator = keyGenerator;
         this.alertColor = alertColor;
     }
 
     /**
+     * Will return true if the name is valid. This means
+     * that it has at least 1 ASCII character.
+     *
+     * @param name The name that has been entered.
+     * @return a {@link boolean} indicating whether or not the name is valid.
+     */
+    private static boolean isValidName(final String name) {
+        return name != null
+                && !name.isEmpty()
+                && !name.trim().isEmpty();
+    }
+
+    /**
+     * Will return true if the name is valid. This means
+     * that it has at least 6 ASCII character.
+     *
+     * @param password The password that has been entered.
+     * @return a {@link boolean} indicating whether or not the password is valid.
+     */
+    private static boolean isValidPassword(final String password) {
+        return password != null
+                && !password.isEmpty()
+                && !password.trim().isEmpty()
+                && password.length() >= MIN_PASSWORD_LENGTH;
+    }
+
+    /**
      * Generates a KeyPair based on the type selected.
      *
      * @param keyType The type of key to generate.
      * @return a {@link KeyPair} with the key type selected
      */
-    Collection<KeyPair> generateKeyPairs(final IKey.Types keyType) {
-        final Collection<KeyPair> keyPairs = new ArrayList<>();
+    List<KeyPair> generateKeyPairs(final IKey.Types keyType) {
```

Comment:
> The fact that I already build it around a list

Well, you might have to rethink your code then


> Also lists are for sure serializable and won't give any trouble writing to file, but I don't know for sure if that isn't the case with collections.

Luckily for you `Collection`s are not serializable, so I'll give you a pass on this one 😉